### PR TITLE
release-1.25: Bump client-go to 1.26.7

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Includes fixes that ensure compatibility with k8s 1.27 clusters

See https://github.com/kubernetes/kubernetes/issues/118361 for context

### CODE REVIEW

File: cmd/contour/serve.go [hunk 1]
```diff
@@ -235,13 +235,11 @@ func NewServer(log logrus.FieldLogger, ctx *serveContext) (*Server, error) {
...
 					// Last-applied-configuration annotation might contain a copy of the complete data.
-					secret.Data = nil
+					secret.Data = map[string][]byte{}
 					secret.SetManagedFields(nil)
 					secret.SetAnnotations(nil)
 
-					// Returning error will avoid handlers from being called.
-					// This does not avoid caching.
-					return nil, fmt.Errorf("ignoring secret %s/%s of type %s", secret.Namespace, secret.Name, secret.Type)
+					return secret, nil
```
Comment (sunjayBhatia) [inline cmd/contour/serve.go:242]: we can't return an error here with the updated handling of the transformer func, see: https://github.com/kubernetes/client-go/commit/fffc68d58e65d023e9783824b1c0fe1c2468c11e

Comment (skriss) [reply cmd/contour/serve.go:242]: Looks consistent with what we did when we upgraded to 1.27 deps (https://github.com/projectcontour/contour/pull/5362), 👍

Comment (sunjayBhatia) [reply cmd/contour/serve.go:242]: contour informers don't sync and this error is repeatedly logged without this

`time="2023-07-26T20:27:34Z" level=error msg="pkg/mod/k8s.io/client-go@v0.26.7/tools/cache/reflector.go:169: Failed to watch *v1.Secret: unable to sync list result: couldn't enqueue object: ignoring secret kube-system/bootstrap-token-abcdef of type bootstrap.kubernetes.io/token\n" caller="reflector.go:140" context=kubernetes error="<nil>"`

**Review by skriss — APPROVED** [review]
LGTM, thanks for sorting this out @sunjayBhatia

**Pull request merged**